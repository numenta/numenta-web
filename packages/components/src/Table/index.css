/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.border {
  composes: br2 from 'tachyons';
  composes: collapse from 'tachyons';
}

.table {
  composes: db from 'tachyons';
  composes: mb4 from 'tachyons';
  composes: mr3 from 'tachyons';
  composes: mr0-ns from 'tachyons';
}
