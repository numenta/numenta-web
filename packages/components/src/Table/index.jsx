// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const Table = ({border, children}) => {
  const classes = [styles.table]
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {border})
  )

  if (border) classes.push(styles.border)

  return (
    <table className={classNames(...classes)}>
      {childrenWithProps}
    </table>
  )
}

Table.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node.isRequired,
}

Table.defaultProps = {
  border: false,
}

export default Table
