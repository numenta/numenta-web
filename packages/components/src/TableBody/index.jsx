// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import styles from './index.css'


/**
 *
 */
const TableBody = ({border, children}) => {
  const childrenWithProps = React.Children.map(children, (child) =>
    React.cloneElement(child, {border})
  )

  return (
    <tbody className={styles.tableBody}>
      {childrenWithProps}
    </tbody>
  )
}

TableBody.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node.isRequired,
}

export default TableBody
