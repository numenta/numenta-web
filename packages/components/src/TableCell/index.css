/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.border {
  composes: b--black-10 from 'tachyons';
  composes: ba from 'tachyons';
}

.tableCell {
  composes: lh-title from 'tachyons';
  composes: pa2 from 'tachyons';
  composes: v-top from 'tachyons';
}
