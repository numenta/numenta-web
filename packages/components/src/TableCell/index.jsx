// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import classNames from 'classnames'
import React from 'react'

import styles from './index.css'


const TableCell = ({border, children, colSpan}) => {
  const classes = [styles.tableCell]

  if (border) classes.push(styles.border)

  return (
    <td className={classNames(...classes)} colSpan={colSpan}>
      {children}
    </td>
  )
}

TableCell.propTypes = {
  border: React.PropTypes.bool,
  children: React.PropTypes.node,
  colSpan: React.PropTypes.number.isRequired,
}

TableCell.defaultProps = {
  colSpan: 1,
}

export default TableCell
