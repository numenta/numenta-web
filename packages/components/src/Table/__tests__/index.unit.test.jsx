import React from 'react'
import renderer from 'react-test-renderer'

import Table from '../../Table'
import TableBody from '../../TableBody'
import TableCell from '../../TableCell'
import TableRow from '../../TableRow'


describe('Table React component', () => {

  it('Renders Bordered correctly', () => {
    const component = renderer.create(
      <Table border={true}>
        <TableBody>
          <TableRow>
            <TableCell>Cell 1</TableCell>
            <TableCell>Cell 2</TableCell>
            <TableCell>Cell 3</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Un-bordered correctly', () => {
    const component = renderer.create(
      <Table border={false}>
        <TableBody>
          <TableRow>
            <TableCell>Cell 1</TableCell>
            <TableCell>Cell 2</TableCell>
            <TableCell>Cell 3</TableCell>
          </TableRow>
        </TableBody>
      </Table>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
