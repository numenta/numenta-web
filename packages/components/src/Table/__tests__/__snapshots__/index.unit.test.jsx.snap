exports[`Table React component Renders Bordered correctly 1`] = `
<table
  className="table border">
  <tbody
    className="tableBody">
    <tr
      className="tableRow">
      <td
        className="tableCell border"
        colSpan={1}>
        Cell 1
      </td>
      <td
        className="tableCell border"
        colSpan={1}>
        Cell 2
      </td>
      <td
        className="tableCell border"
        colSpan={1}>
        Cell 3
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table React component Renders Un-bordered correctly 1`] = `
<table
  className="table">
  <tbody
    className="tableBody">
    <tr
      className="tableRow">
      <td
        className="tableCell"
        colSpan={1}>
        Cell 1
      </td>
      <td
        className="tableCell"
        colSpan={1}>
        Cell 2
      </td>
      <td
        className="tableCell"
        colSpan={1}>
        Cell 3
      </td>
    </tr>
  </tbody>
</table>
`;
